# Mobile Application analysis

This project is about analyze mobile application data on both IOS and android that attract more user in order to make more revenue from in-ads advertisement.Our goal for this project is to analyze data to help our developers understand what type of apps are likely to attract more users.

First, read data from files

In [ ]:
open_IOS_file = open('C:/Users/dowre/GIT/Mobile app project/AppleStore.csv', encoding="utf8")
open_Android_file = open('C:/Users/dowre/GIT/Mobile app project/googleplaystore.csv', encoding="utf8")

In [2]:
from csv import reader
### The Google Play data set ###

read_file = reader(open_Android_file)
android = list(read_file)
android_header = android[0]
android = android[1:]

### The App Store data set ###

read_file = reader(open_IOS_file)
ios = list(read_file)
ios_header = ios[0]
ios = ios[1:]


Build a function that allows you to see some specific part of the file after modification.

In [3]:
def explore_data(dataset,start,end,rows_and_columns=False):
    dataset_slice = dataset[start:end]
    for row in dataset_slice:
        print(row)
        print('\n') #add a new (empty) line after each row 
    if rows_and_columns:
        print('Number of rows:',len(dataset))
        print('Number of columns:', len(dataset[0]))

Check out a few first rows of data in both files. Also check total number of rows and columns.

In [4]:
explore_data(ios,0,2,rows_and_columns=True)
explore_data(android,0,2,rows_and_columns=True)

['284882215', 'Facebook', '389879808', 'USD', '0', '2974676', '212', '3.5', '3.5', '95', '4+', 'Social Networking', '37', '1', '29', '1', '']


['389801252', 'Instagram', '113954816', 'USD', '0', '2161558', '1289', '4.5', '4', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1', '']


Number of rows: 7197
Number of columns: 17
['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


Number of rows: 10841
Number of columns: 13


In [5]:
len(android)

10841

In [6]:
len(ios)

7197

**Check out column name of both files**

Please see the following link for more descriptions of each column names.
- For [IOS](https://www.kaggle.com/ramamet4/app-store-apple-data-set-10k-apps) 
- For [Android](https://www.kaggle.com/lava18/google-play-store-apps)

In [7]:
print(ios_header)

['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic', '']


In [8]:
print(android_header)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


## Data Cleaning

### 1. Explore each row length to verify missing value in Android and IOS data set.

Define the function to check length in each row

In [9]:
def lenght_check(row,store):
    length = 0
    if store == 'google':
        #selected_store = 'google'
        length = 13
    elif store == 'apple':
        #selected_store = 'apple'
        length =  16
    else:
        return print("Wrong App Store name. Only 'google' and  'apple' are valid")
    if len(row) != length:
        return False
    return True

In [10]:
lenght_check(ios[5645], 'apple')

False

In [11]:
print(ios[5645])

['454136236', 'STEINS', 'GATE', '1551412224', 'USD', '24.99', '16', '0', '4', '0', '1.71', '12+', 'Games', '37', '0', '1', '1']


In [12]:
len(ios[5645])

17

In [13]:
lenght_check(android[10472], 'google')

False

In [14]:
len(android[10472])

12

In [15]:
lenght_check(ios[1], 'apple')

False

In [16]:
type(ios[1][4])

str

In [17]:
print(ios[1])

['389801252', 'Instagram', '113954816', 'USD', '0', '2161558', '1289', '4.5', '4', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1', '']


In [18]:
len(ios[1])

17

In [19]:
def row_check(data_set,store):
    #selected_store =''
    length = 0
    if store == 'google':
        #selected_store = 'google'
        length = 13
    elif store == 'apple':
        #selected_store = 'apple'
        length =  16
    else:
        return print("Wrong App Store name. Only 'google' and  'apple' are valid")
    for app in data_set:
        incomplete_app_index = 0
        incomplete_app_index_list =[]
        if len(app) != length:
            incomplete_app_index = data_set.index(app)
            incomplete_app_index_list.append(incomplete_app_index)
    return incomplete_app_index_list

Apply the function to andriod data set

In [20]:
row_check(android,'google')

[]

There is one missing data row which is number 10472. Check out the missing data row.

In [21]:
print(android[10472])

['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']


In [22]:
len(android[10472])

12

Delete the row that contain missing value

In [23]:
#Do not run more than one time
del android[10472]

Check deleting result

In [24]:
row_check(android,'google')

[]

There is no row with missing value.

Check the length of the file. 

In [25]:
len(android)

10840

The lenght is decreased by one

Apply the function to IOS data set

In [26]:
row_check(ios,'apple')

[7196]

There is no row that contain missing value in IOS dataset

### 2. Explore the duplicate data 

Check the example duplicated data in Android file

In [27]:
for app in android:
    name = app[0]
    if name == 'Instagram':
        print(app)

['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577446', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66509917', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']


Create function to check the duplicate app in each file.

In [28]:
def check_duplicate(dataset,store):
    duplicate_app = []
    unique_app =[]
    app_index= []
    if store == 'google' or  store == 'apple':
        for app in dataset:
            app_name = app[0]
            if app_name in unique_app:
                duplicate_app.append(app)
                duplicate_app_index = dataset.index(app)
                app_index.append(duplicate_app_index)
            else:
                unique_app.append(app_name)
    # For app store ,ID column is used for duplicated check instead of application name
#     elif store == 'apple':
#         for app in dataset:
#             app_name = app[1]
#             if app_name in unique_app:
#                 duplicate_app.append(app)
#                 duplicate_app_index = dataset.index(app)
#                 app_index.append(duplicate_app_index)
#             else:
#                 unique_app.append(app_name)
    else:
        return print("Wrong App Store name. Only 'google' and  'apple' are valid")
    return print('Number of duplicate app in '+store ,len(duplicate_app)) ,print('Number of unique app in '+store ,len(unique_app)), print(app_index)    

In [29]:
android_duplicate = check_duplicate(android, 'google')

Number of duplicate app in google 1181
Number of unique app in google 9659
[222, 204, 193, 213, 253, 204, 237, 238, 193, 240, 241, 242, 243, 244, 245, 246, 247, 248, 249, 250, 251, 252, 253, 254, 228, 285, 230, 231, 232, 233, 234, 235, 192, 293, 294, 243, 336, 382, 383, 385, 386, 390, 392, 393, 359, 369, 408, 410, 411, 412, 372, 415, 417, 419, 420, 388, 385, 436, 451, 452, 464, 383, 389, 467, 391, 390, 399, 402, 403, 407, 483, 490, 485, 486, 488, 487, 520, 495, 489, 491, 518, 503, 516, 525, 494, 500, 502, 564, 565, 566, 568, 571, 574, 575, 576, 579, 580, 534, 528, 590, 596, 662, 663, 664, 603, 610, 496, 506, 670, 605, 608, 673, 674, 602, 589, 611, 679, 607, 528, 604, 613, 606, 609, 529, 600, 501, 697, 702, 700, 734, 741, 702, 700, 784, 740, 742, 741, 756, 789, 749, 793, 750, 796, 784, 801, 737, 807, 740, 824, 784, 800, 803, 801, 832, 717, 835, 809, 766, 824, 843, 844, 740, 835, 801, 883, 903, 855, 888, 931, 932, 933, 914, 912, 913, 917, 938, 919, 921, 916, 924, 920, 918, 950, 922, 958,

In [30]:
ios_duplicate = check_duplicate(ios, 'apple')

Number of duplicate app in apple 0
Number of unique app in apple 7197
[]


**Define functions to create new clean file with no duplicated data.**
- For the duplicated data , The row that will be keep in the clean file is the one with the highest review number. 

There are 2 functions.
1. Creating dictionary to collect unique app name with its highest review number.
2. Creating new clean application list file base on the dictionary.

In [31]:
#Create review_max data dictionary
def review_max_dict(dataset,store): 
    review_max ={}
    for app in dataset:
        #for data in google play store
        if store == 'google':
            name = app[0]
            n_reviews = float(app[3])
            if name in review_max and review_max[name] < n_reviews :
                review_max[name] = n_reviews
            elif name not in review_max:
                review_max[name] = n_reviews
        #for data in apple store
        elif store == 'apple':
            app_id = app[0]
            n_reviews = float(app[5])
            if app_id in review_max and review_max[app_id] < n_reviews :
                review_max[app_id] = n_reviews
            elif app_id not in review_max:
                review_max[app_id] = n_reviews
        elif store != 'apple'or store != 'google':
            print("Wrong App Store name. Only 'google' and  'apple' are valid")
    return review_max

In [32]:
#Create new list with non duplicate value 
def clean_app(dataset,review_max, store):
    clean_list = []
    already_added = []
    for app in dataset:
        if store == 'google':
            name = app[0]
            n_reviews = float(app[3])
            if (n_reviews == review_max[name]) and (name not in already_added):
                clean_list.append(app)
                already_added.append(name)
        elif store == 'apple':
            app_id = app[0]
            n_reviews = float(app[5])
            if (n_reviews == review_max[app_id]) and (app_id not in already_added):
                clean_list.append(app)
                already_added.append(app_id)
        else:
            print('Wrong App Store name. Only "google" and  "apple" are valid')
    return clean_list 

Apply the function to Android file

In [33]:
android_dict = review_max_dict(android,'google')

In [34]:
type(android_dict)

dict

In [35]:
print(len((android_dict)))

9659


In [36]:
new_android = clean_app(android,android_dict,'google')

In [37]:
len(new_android)

9659

In [38]:
explore_data(new_android,0,3,True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows: 9659
Number of columns: 13


 This function doesn't need to be run for IOS file according to there is no duplicated data in the file.

### 3. Filter out app name that is not English language.

Because our company build applications in English so we will pay attention to customers who is english language user.

- Define function to check whether the app name is English language
- Create new list that contain only English application name.

In [39]:
def is_english(name):
    count = 0
    for string in name:
        if ord(string) > 127:
            count += 1
    if count > 3:
        return False
    return True

In [40]:
is_english('Docs To Go™ Free Office Suite')

True

In [41]:
is_english('Instachat 😜')

True

In [42]:
is_english('爱奇艺PPS -《欢乐颂2》电视剧热播')

False

In [43]:
def english_app(dataset,store):
    english_app_name = []
    for app in dataset:
        if store == 'google':
            app_name = app[0]
        elif store == 'apple':
            app_name = app[1]
        else:
            return print('Wrong App Store name. Only "google" and  "apple" are valid')
        if is_english(app_name):
            english_app_name.append(app)  
    return english_app_name

Apply with New_Android file

In [44]:
android_english= english_app(new_android, 'google')

In [45]:
explore_data(android_english,0,3,True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows: 9614
Number of columns: 13


Apply with ios file

In [46]:
IOS_english= english_app(ios, 'apple')

In [47]:
explore_data(IOS_english,0,3,True)

['284882215', 'Facebook', '389879808', 'USD', '0', '2974676', '212', '3.5', '3.5', '95', '4+', 'Social Networking', '37', '1', '29', '1', '']


['389801252', 'Instagram', '113954816', 'USD', '0', '2161558', '1289', '4.5', '4', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1', '']


['529479190', 'Clash of Clans', '116476928', 'USD', '0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1', '']


Number of rows: 6183
Number of columns: 17


### 4. Filter only free application

Create function that check each application's price to identify free app.

In [48]:
android_free = []
ios_free = []

for app in android_english:
    app_price = app[7]
    if app[7] == '0':
        android_free.append(app)

for app in IOS_english:
    app_price = app[4]
    if app[4] == '0':
        ios_free.append(app)

        
print("Total Android apps are", len(android_free))
print("Total ios apps are", len(ios_free))

Total Android apps are 8864
Total ios apps are 3222


In [49]:
explore_data(android_free,0,3,True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows: 8864
Number of columns: 13


In [50]:
explore_data(ios_free,0,3,True)

['284882215', 'Facebook', '389879808', 'USD', '0', '2974676', '212', '3.5', '3.5', '95', '4+', 'Social Networking', '37', '1', '29', '1', '']


['389801252', 'Instagram', '113954816', 'USD', '0', '2161558', '1289', '4.5', '4', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1', '']


['529479190', 'Clash of Clans', '116476928', 'USD', '0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1', '']


Number of rows: 3222
Number of columns: 17


Now we have 8,864 total number of row for Android file and 3,222 rows for IOS file.

## Data Analysis

Our company's aim is to determine the kinds of apps that are likely to attract more users because Our company's revenue is highly influenced by the number of people using our apps.

To minimize risks and overhead, our validation strategy for an app idea is comprised of three steps:

1. Build a minimal Android version of the app, and add it to Google Play.
2. If the app has a good response from users, we develop it further.
3. If the app is profitable after six months, we build an iOS version of the app and add it to the App Store.

### 1. Find out what is the most popular application genre in both market.
 
First we want to know what genre of free application that are the most popular in both markets.

In [51]:
def freq_genre (dataset,index):
    frequency_genre = {}
    for app in dataset:
        app_genre = app[index]
        if app_genre not in frequency_genre:
            frequency_genre[app_genre] = 1
        else: 
            frequency_genre[app_genre] += 1
    return frequency_genre

In [52]:
frequency_android_category = freq_genre(android_free, 1)#Use Category column
frequency_android_category

{'ART_AND_DESIGN': 57,
 'AUTO_AND_VEHICLES': 82,
 'BEAUTY': 53,
 'BOOKS_AND_REFERENCE': 190,
 'BUSINESS': 407,
 'COMICS': 55,
 'COMMUNICATION': 287,
 'DATING': 165,
 'EDUCATION': 103,
 'ENTERTAINMENT': 85,
 'EVENTS': 63,
 'FINANCE': 328,
 'FOOD_AND_DRINK': 110,
 'HEALTH_AND_FITNESS': 273,
 'HOUSE_AND_HOME': 73,
 'LIBRARIES_AND_DEMO': 83,
 'LIFESTYLE': 346,
 'GAME': 862,
 'FAMILY': 1676,
 'MEDICAL': 313,
 'SOCIAL': 236,
 'SHOPPING': 199,
 'PHOTOGRAPHY': 261,
 'SPORTS': 301,
 'TRAVEL_AND_LOCAL': 207,
 'TOOLS': 750,
 'PERSONALIZATION': 294,
 'PRODUCTIVITY': 345,
 'PARENTING': 58,
 'WEATHER': 71,
 'VIDEO_PLAYERS': 159,
 'NEWS_AND_MAGAZINES': 248,
 'MAPS_AND_NAVIGATION': 124}

In [53]:
frequency_android_genre = freq_genre(android_free, -4)#Use genre column
frequency_android_genre

{'Art & Design': 53,
 'Art & Design;Creativity': 6,
 'Auto & Vehicles': 82,
 'Beauty': 53,
 'Books & Reference': 190,
 'Business': 407,
 'Comics': 54,
 'Comics;Creativity': 1,
 'Communication': 287,
 'Dating': 165,
 'Education': 474,
 'Education;Creativity': 4,
 'Education;Education': 30,
 'Education;Pretend Play': 5,
 'Education;Brain Games': 3,
 'Entertainment': 538,
 'Entertainment;Brain Games': 7,
 'Entertainment;Creativity': 3,
 'Entertainment;Music & Video': 15,
 'Events': 63,
 'Finance': 328,
 'Food & Drink': 110,
 'Health & Fitness': 273,
 'House & Home': 73,
 'Libraries & Demo': 83,
 'Lifestyle': 345,
 'Lifestyle;Pretend Play': 1,
 'Card': 40,
 'Arcade': 164,
 'Puzzle': 100,
 'Racing': 88,
 'Sports': 307,
 'Casual': 156,
 'Simulation': 181,
 'Adventure': 60,
 'Trivia': 37,
 'Action': 275,
 'Word': 23,
 'Role Playing': 83,
 'Strategy': 81,
 'Board': 34,
 'Music': 18,
 'Action;Action & Adventure': 9,
 'Casual;Brain Games': 12,
 'Educational;Creativity': 3,
 'Puzzle;Brain Games':

In [54]:
frequency_ios_genre = freq_genre(ios_free, -6)
frequency_ios_genre

{'Social Networking': 106,
 'Photo & Video': 160,
 'Games': 1874,
 'Music': 66,
 'Reference': 18,
 'Health & Fitness': 65,
 'Weather': 28,
 'Utilities': 81,
 'Travel': 40,
 'Shopping': 84,
 'News': 43,
 'Navigation': 6,
 'Lifestyle': 51,
 'Entertainment': 254,
 'Food & Drink': 26,
 'Sports': 69,
 'Book': 14,
 'Finance': 36,
 'Education': 118,
 'Productivity': 56,
 'Business': 17,
 'Catalogs': 4,
 'Medical': 6}

Transform the number into percentage

In [55]:
total_android_app = len(android_free)
total_ios_app = len(ios_free)
print ("Total number of genre for Android is : ", total_android_app)
print ("Total number of genre for IOS is : ", total_ios_app)


Total number of genre for Android is :  8864
Total number of genre for IOS is :  3222


In [56]:
def percentage_genre (dataset,total):
    percentage_dict= {}
    for key in dataset:
        percentage = (dataset[key]/total)*100
        percentage_dict[key] = percentage
    return percentage_dict

# for key in frequency_genre_ios:
#     percentage = (frequency_genre_ios[key]/total_ios_app)*100
#     percentage_dict_IOS[key] = percentage

In [57]:
percentage_dict_Android_cat = percentage_genre(frequency_android_category,total_android_app )
percentage_dict_Android_genre = percentage_genre(frequency_android_genre,total_android_app )
percentage_dict_IOS = percentage_genre(frequency_ios_genre,total_ios_app )

Sort dictionary of Android **Category** by percentage value 

In [58]:
import operator
sort_percentage_android_category = sorted(percentage_dict_Android_cat.items(), key = operator.itemgetter(1), reverse = True)
sort_percentage_android_category

[('FAMILY', 18.907942238267147),
 ('GAME', 9.724729241877256),
 ('TOOLS', 8.461191335740072),
 ('BUSINESS', 4.591606498194946),
 ('LIFESTYLE', 3.9034296028880866),
 ('PRODUCTIVITY', 3.892148014440433),
 ('FINANCE', 3.7003610108303246),
 ('MEDICAL', 3.531137184115524),
 ('SPORTS', 3.395758122743682),
 ('PERSONALIZATION', 3.3167870036101084),
 ('COMMUNICATION', 3.2378158844765346),
 ('HEALTH_AND_FITNESS', 3.0798736462093865),
 ('PHOTOGRAPHY', 2.944494584837545),
 ('NEWS_AND_MAGAZINES', 2.7978339350180503),
 ('SOCIAL', 2.6624548736462095),
 ('TRAVEL_AND_LOCAL', 2.33528880866426),
 ('SHOPPING', 2.2450361010830324),
 ('BOOKS_AND_REFERENCE', 2.1435018050541514),
 ('DATING', 1.861462093862816),
 ('VIDEO_PLAYERS', 1.7937725631768955),
 ('MAPS_AND_NAVIGATION', 1.3989169675090252),
 ('FOOD_AND_DRINK', 1.2409747292418771),
 ('EDUCATION', 1.1620036101083033),
 ('ENTERTAINMENT', 0.9589350180505415),
 ('LIBRARIES_AND_DEMO', 0.9363718411552346),
 ('AUTO_AND_VEHICLES', 0.9250902527075812),
 ('HOUSE_AN

For Android ,the percentage doesn't goes toward one category too much. It is more scatter than IOS.

From the result of percentage of each Android **category** above, the most popular category are Family which is the application for children which can break down into different kinds of genre e.g. education, entertainment, music , diffetent type of games and etc.

Sort dictionary of Android **genre** by percentage value

In [59]:
sort_percentage_android_genre = sorted(percentage_dict_Android_genre.items(), key = operator.itemgetter(1), reverse = True)
sort_percentage_android_genre

[('Tools', 8.449909747292418),
 ('Entertainment', 6.069494584837545),
 ('Education', 5.347472924187725),
 ('Business', 4.591606498194946),
 ('Lifestyle', 3.892148014440433),
 ('Productivity', 3.892148014440433),
 ('Finance', 3.7003610108303246),
 ('Medical', 3.531137184115524),
 ('Sports', 3.463447653429603),
 ('Personalization', 3.3167870036101084),
 ('Communication', 3.2378158844765346),
 ('Action', 3.1024368231046933),
 ('Health & Fitness', 3.0798736462093865),
 ('Photography', 2.944494584837545),
 ('News & Magazines', 2.7978339350180503),
 ('Social', 2.6624548736462095),
 ('Travel & Local', 2.3240072202166067),
 ('Shopping', 2.2450361010830324),
 ('Books & Reference', 2.1435018050541514),
 ('Simulation', 2.0419675090252705),
 ('Dating', 1.861462093862816),
 ('Arcade', 1.8501805054151623),
 ('Video Players & Editors', 1.7712093862815883),
 ('Casual', 1.7599277978339352),
 ('Maps & Navigation', 1.3989169675090252),
 ('Food & Drink', 1.2409747292418771),
 ('Puzzle', 1.128158844765343)

From the result of percentage of each Android **genre** above, it is obviously different from category. Tools genre is the most popular followed by entertainment and education.

Compare Tools between category and genre
Category  : ('TOOLS', 8.461191335740072)
genre : ('Tools', 8.449909747292418)

The percentage are almost equal.


Compare Entertainment and Education between category and genre, 

for category ,('EDUCATION', 1.1620036101083033),  ('ENTERTAINMENT', 0.9589350180505415)

for genre ,

Education
('Education', 5.347472924187725),  ('Educational;Education', 0.39485559566787), ('Educational', 0.3722924187725632), ('Education;Education', 0.33844765342960287),('Educational;Pretend Play', 0.09025270758122744),('Educational;Brain Games', 0.06768953068592057), ('Education;Pretend Play', 0.056407942238267145),
('Education;Creativity', 0.04512635379061372),('Education;Brain Games', 0.033844765342960284),('Educational;Creativity', 0.033844765342960284),
('Educational;Action & Adventure', 0.033844765342960284),('Education;Action & Adventure', 0.033844765342960284),('Education;Music & Video', 0.033844765342960284),

Total percentage : 6.87

Entertainment
('Entertainment', 6.069494584837545),('Entertainment;Music & Video', 0.16922382671480143), ('Entertainment;Brain Games', 0.078971119133574),('Entertainment;Creativity', 0.033844765342960284),('Entertainment;Action & Adventure', 0.033844765342960284),('Entertainment;Pretend Play', 0.02256317689530686),('Entertainment;Education', 0.01128158844765343)

Total percentage : 6.41

Education and Entertainment genre are a lot higher than category because most of them are genre of Family category. The percentage of Family category is 18.907942238267147 which is the highest number. 
If we try to subtract Education genre that is under Education category which is 6.87 -  1.16 = 5.71 . This number can represent the percentage of Education genre under Family category.
Do the same with Entertainment. 6.41 - 0.96 = 5.45. This number can represent the percentage of Education genre under Family category.

The percentage of Education and Entertainment genre under Family category are 
 - Education genre is 5.71
 - Entertainment genre is 5.45

The summary of Education and Entertainment genre in Family category is 11.16 which is higher than Tools.

The conclusion is the most popular genre are under Family category as well. 
As you can see that Education is a bit higher than Entertainment. There are still many genres under Family category for example ('Simulation', 2.0419675090252705) which is mostly under Family cateory and some are under Games. Another example is (Puzzle', 1.128158844765343) which is mostly under Family cateory and some are under Games as well which both of them considered to be in Entertainment genre.

For Android, the most popular genre is Entertainment genre in Family category.

Sort dictionary of **IOS** genre by percentage value

In [60]:
sort_percentage_IOS = sorted(percentage_dict_IOS.items(), key = operator.itemgetter(1), reverse = True)
sort_percentage_IOS

[('Games', 58.16263190564867),
 ('Entertainment', 7.883302296710118),
 ('Photo & Video', 4.9658597144630665),
 ('Education', 3.662321539416512),
 ('Social Networking', 3.2898820608317814),
 ('Shopping', 2.60707635009311),
 ('Utilities', 2.5139664804469275),
 ('Sports', 2.1415270018621975),
 ('Music', 2.0484171322160147),
 ('Health & Fitness', 2.0173805090006205),
 ('Productivity', 1.7380509000620732),
 ('Lifestyle', 1.5828677839851024),
 ('News', 1.3345747982619491),
 ('Travel', 1.2414649286157666),
 ('Finance', 1.1173184357541899),
 ('Weather', 0.8690254500310366),
 ('Food & Drink', 0.8069522036002483),
 ('Reference', 0.5586592178770949),
 ('Business', 0.5276225946617008),
 ('Book', 0.4345127250155183),
 ('Navigation', 0.186219739292365),
 ('Medical', 0.186219739292365),
 ('Catalogs', 0.12414649286157665)]

From the results above, for IOS ,the most popular free english app genre is Games. The top three are in Entertainment purpose. If we choose only entertainment genre app and sum up the perentage it will be equal to 78.46% (from Games,Entertainment,Photo & Video,Social Networking,Sports,Music genre). Only 6 genres in the store takes very large number of percentage. It indicates that entertainment apps attract more users than others.

However, the number of application in the market doesn't reflect the demand of users.

### 2. Find out what is the most popular genre of application that has been installed or used.

One way to find out what genres are the most popular (have the most users) is to calculate the average number of installs for each app genre. For the Google Play data set, we can find this information in the Installs column, but this information is missing for the App Store data set. As a workaround, we'll take the total number of user ratings as a proxy, which we can find in the rating_count_tot app.

Let's start with calculating the average number of user ratings per app genre on the App Store. To do that, we'll need to:

- Isolate the apps of each genre. 
- Sum up the user ratings for the apps of that genre.
- Divide the sum by the number of apps belonging to that genre (not by the total number of apps).

For **IOS**

To isolate the apps of each genre , we can use the function which is defined above.

In [62]:
frequency_ios_genre

{'Social Networking': 106,
 'Photo & Video': 160,
 'Games': 1874,
 'Music': 66,
 'Reference': 18,
 'Health & Fitness': 65,
 'Weather': 28,
 'Utilities': 81,
 'Travel': 40,
 'Shopping': 84,
 'News': 43,
 'Navigation': 6,
 'Lifestyle': 51,
 'Entertainment': 254,
 'Food & Drink': 26,
 'Sports': 69,
 'Book': 14,
 'Finance': 36,
 'Education': 118,
 'Productivity': 56,
 'Business': 17,
 'Catalogs': 4,
 'Medical': 6}

Sum up the user ratings for the apps of that genre

Divide the sum by the number of apps belonging to that genre (not by the total number of apps).

In [63]:
avg_user_genre ={}
for genre in frequency_ios_genre:
    rating_count = 0
    for app in ios_free:
        genre_app = app[-6]
        if genre_app == genre:
            rating_count += float(app[5])
    len_genre = frequency_ios_genre[genre]
    avg_number = rating_count/len_genre
    print( 'Average user number in', genre, 'genre is :' ,avg_number )
    avg_user_genre[genre] = avg_number

Average user number in Social Networking genre is : 71548.34905660378
Average user number in Photo & Video genre is : 28441.54375
Average user number in Games genre is : 22788.6696905016
Average user number in Music genre is : 57326.530303030304
Average user number in Reference genre is : 74942.11111111111
Average user number in Health & Fitness genre is : 23298.015384615384
Average user number in Weather genre is : 52279.892857142855
Average user number in Utilities genre is : 18684.456790123455
Average user number in Travel genre is : 28243.8
Average user number in Shopping genre is : 26919.690476190477
Average user number in News genre is : 21248.023255813954
Average user number in Navigation genre is : 86090.33333333333
Average user number in Lifestyle genre is : 16485.764705882353
Average user number in Entertainment genre is : 14029.830708661417
Average user number in Food & Drink genre is : 33333.92307692308
Average user number in Sports genre is : 23008.898550724636
Average use

In [64]:
avg_user_genre

{'Social Networking': 71548.34905660378,
 'Photo & Video': 28441.54375,
 'Games': 22788.6696905016,
 'Music': 57326.530303030304,
 'Reference': 74942.11111111111,
 'Health & Fitness': 23298.015384615384,
 'Weather': 52279.892857142855,
 'Utilities': 18684.456790123455,
 'Travel': 28243.8,
 'Shopping': 26919.690476190477,
 'News': 21248.023255813954,
 'Navigation': 86090.33333333333,
 'Lifestyle': 16485.764705882353,
 'Entertainment': 14029.830708661417,
 'Food & Drink': 33333.92307692308,
 'Sports': 23008.898550724636,
 'Book': 39758.5,
 'Finance': 31467.944444444445,
 'Education': 7003.983050847458,
 'Productivity': 21028.410714285714,
 'Business': 7491.117647058823,
 'Catalogs': 4004.0,
 'Medical': 612.0}

In [69]:
sorted(avg_user_genre.items(), key= operator.itemgetter(1),reverse = True)

[('Navigation', 86090.33333333333),
 ('Reference', 74942.11111111111),
 ('Social Networking', 71548.34905660378),
 ('Music', 57326.530303030304),
 ('Weather', 52279.892857142855),
 ('Book', 39758.5),
 ('Food & Drink', 33333.92307692308),
 ('Finance', 31467.944444444445),
 ('Photo & Video', 28441.54375),
 ('Travel', 28243.8),
 ('Shopping', 26919.690476190477),
 ('Health & Fitness', 23298.015384615384),
 ('Sports', 23008.898550724636),
 ('Games', 22788.6696905016),
 ('News', 21248.023255813954),
 ('Productivity', 21028.410714285714),
 ('Utilities', 18684.456790123455),
 ('Lifestyle', 16485.764705882353),
 ('Entertainment', 14029.830708661417),
 ('Business', 7491.117647058823),
 ('Education', 7003.983050847458),
 ('Catalogs', 4004.0),
 ('Medical', 612.0)]

From the result, Number of rating of Navigation, Reference and Social Networking are top three highest. Games and Entertainment are not so high compare to others. These number can represent the number of real users that use the application. 
In order to come up with conclusion I would like to have more data e.g. average time that user spend in each applications. Unfortunately we don't have it.
My analysis from data that we have so far is the application genre that are interesting to create are Reference, Social Networking or Music genre. 

The reason are because those genre has 
- high number of rating 
- most liking to have long user spending time in the app. 
- user have to read or listen to the content inside the app so there is higher chance to see advertisement.

Even Navigation has the highest rating count but user may not need to pay attention to the content at all time especially when they use it while driving.

However we should take into consideration about the existing competitors that already own high portion of market share.

Let's see the percentage of user review in each existing application in Social Networking genre.

In [79]:
total_rating_SN = 0
for app in ios_free:
    if app[-6] == 'Social Networking':
        total_rating_SN += float(app[5])
        
for app in ios_free:
    if app[-6] == 'Social Networking':
        percent_rating_count = (float(app[5])*100)/total_rating_SN
        #print (app[1], ':',app[5], ' percent :' ,percent_rating_count ,'%' )
        print (app[1], ':' ,percent_rating_count ,'%' )

Facebook : 39.222402056928125 %
Pinterest : 13.997976035468824 %
Skype for iPhone : 4.925011125212203 %
Messenger : 4.634232689993902 %
Tumblr : 4.407799185799285 %
WhatsApp Messenger : 3.791986550854581 %
Kik : 3.4409374845483165 %
ooVoo – Free Video Call, Text and Voice : 2.3404281970563514 %
TextNow - Unlimited Text + Calls : 2.1751091918975494 %
Viber Messenger – Text & Call : 2.1656947901043297 %
Followers - Social Analytics For Instagram : 1.4870271784813673 %
MeetMe - Chat and Meet New People : 1.2799367099039112 %
We Heart It - Fashion, wallpapers, quotes, tattoos : 1.1921480724539746 %
InsTrack for Instagram - Analytics Plus More : 1.1278163268669754 %
Tango - Free Video Call, Voice and Chat : 0.994340151302886 %
LinkedIn : 0.9474527384503816 %
Match™ - #1 Dating App. : 0.7998154038864075 %
Skype for iPad : 0.7932754272905576 %
POF - Best Dating App for Conversations : 0.6941077579813096 %
Timehop : 0.652810970283322 %
Find My Family, Friends & iPhone - Life360 Locator : 0.578

For Social networking , Facebook and Pinterest take about 52% of the total app in genre. Majority of the reamaining apps has less than 1 percent of number of user rating.

Let's see the percentage of user review in each existing application in Music genre.

In [80]:
total_rating_MS = 0
for app in ios_free:
    if app[-6] == 'Music':
        total_rating_MS += float(app[5])
        
for app in ios_free:
    if app[-6] == 'Music':
        percent_rating_count = (float(app[5])*100)/total_rating_MS
        print (app[1], ':' ,percent_rating_count ,'%' )

Pandora - Music & Radio : 29.78363447459807 %
Spotify Music : 23.22059356408834 %
Shazam - Discover music, artists, videos & lyrics : 10.649387308377765 %
iHeartRadio – Free Music & Radio Stations : 7.750073938477372 %
SoundCloud - Music & Audio : 3.587740722934619 %
Magic Piano by Smule : 3.4807248534511626 %
Smule Sing! : 3.1535454391919124 %
TuneIn Radio - MLB NBA Audiobooks Podcasts Music : 2.9184224026582437 %
Amazon Music : 2.807812026321305 %
SoundHound Song Search & Music Player : 2.1831871699363905 %
Sonos Controller : 1.292568806393782 %
Bandsintown Concerts : 0.8152394404092874 %
Karaoke - Sing Karaoke, Unlimited Songs! : 0.7560622283140891 %
My Mixtapez Music : 0.6947441702252725 %
Sing Karaoke Songs Unlimited with StarMaker : 0.6931847885756001 %
Ringtones for iPhone & Ringtone Maker : 0.6714063058750893 %
Musi - Unlimited Music For YouTube : 0.6658559644101533 %
AutoRap by Smule : 0.4810824540226893 %
Spinrilla - Mixtapes For Free : 0.39785376224610164 %
Napster - Top Mus

For Music, Pandora and Spotify take about 52% of the total app in the genre as well. 

Let's see the percentage of user review in each existing application in Reference genre.

In [81]:
total_rating_RF = 0
for app in ios_free:
    if app[-6] == 'Reference':
        total_rating_RF += float(app[5])
        
for app in ios_free:
    if app[-6] == 'Reference':
        percent_rating_count = (float(app[5])*100)/total_rating_RF
        print (app[1], ':' ,percent_rating_count ,'%' )

Bible : 73.08752385174334 %
Dictionary.com Dictionary & Thesaurus : 14.829742660631391 %
Dictionary.com Dictionary & Thesaurus for iPad : 4.016062768447942 %
Google Translate : 1.985680799550468 %
Muslim Pro: Ramadan 2017 Prayer Times, Azan, Quran : 1.3653501443336264 %
New Furniture Mods - Pocket Wiki & Game Tools for Minecraft PC Edition : 1.3038211716005983 %
Merriam-Webster Dictionary : 1.2490381464804687 %
Night Sky : 0.8986195270720067 %
City Maps for Minecraft PE - The Best Maps for Minecraft Pocket Edition (MCPE) : 0.6327105810558965 %
LUCKY BLOCK MOD ™ for Minecraft PC Edition - The Best Pocket Wiki & Mods Installer Tools : 0.34789815546518127 %
GUNS MODS for Minecraft PC Edition - Mods Tools : 0.11097454479679872 %
Guides for Pokémon GO - Pokemon GO News and Cheats : 0.061232447563230284 %
WWDC : 0.05648804484646668 %
Horror Maps for Minecraft PE - Download The Scariest Maps for Minecraft Pocket Edition (MCPE) Free : 0.0532262679786917 %
VPN Express : 0.0010378380942920387 %


For Reference, Bible and dictionary take around 87% of of the total app in the genre.

As you can see that , in each genre there are a couple apps that hold a huge percentage than the other remaining apps. 
The key to attract more user is to focus on specific target group or make something different. The one that have higher chance might be Reference because there still be not too many of competitors.

**For Android** 

Repeat the same process with IOS but there are some different.
We have data about the number of installs for the Google Play market, so we should be able to get a clearer picture about genre popularity. However, the install numbers don't seem precise enough — we can see that most values are open-ended (100+, 1,000+, 5,000+, etc.) We're going to leave the numbers as they are, which means that we'll consider that an app with 100,000+ installs has 100,000 installs, and an app with 1,000,000+ installs has 1,000,000 installs, and so on.

In order to use that number we have to convert string to float. As you can see the string contain , and + sign so we have to replace it before convert.

- Isolate the apps of each genre.

There are Category and Genre for Android , we decided to use Category because it gives you overview idea better than genre which is may be too specific.

In [83]:
frequency_android_category

{'ART_AND_DESIGN': 57,
 'AUTO_AND_VEHICLES': 82,
 'BEAUTY': 53,
 'BOOKS_AND_REFERENCE': 190,
 'BUSINESS': 407,
 'COMICS': 55,
 'COMMUNICATION': 287,
 'DATING': 165,
 'EDUCATION': 103,
 'ENTERTAINMENT': 85,
 'EVENTS': 63,
 'FINANCE': 328,
 'FOOD_AND_DRINK': 110,
 'HEALTH_AND_FITNESS': 273,
 'HOUSE_AND_HOME': 73,
 'LIBRARIES_AND_DEMO': 83,
 'LIFESTYLE': 346,
 'GAME': 862,
 'FAMILY': 1676,
 'MEDICAL': 313,
 'SOCIAL': 236,
 'SHOPPING': 199,
 'PHOTOGRAPHY': 261,
 'SPORTS': 301,
 'TRAVEL_AND_LOCAL': 207,
 'TOOLS': 750,
 'PERSONALIZATION': 294,
 'PRODUCTIVITY': 345,
 'PARENTING': 58,
 'WEATHER': 71,
 'VIDEO_PLAYERS': 159,
 'NEWS_AND_MAGAZINES': 248,
 'MAPS_AND_NAVIGATION': 124}

- Sum up the number of installed for the apps of that genre.
- Divide the sum by the number of apps belonging to that genre (not by the total number of apps).

In [86]:
avg_install_genre ={}
for genre in frequency_android_category:
    install_count = 0
    for app in android_free:
        genre_app = app[1]
        if genre_app == genre:
            n_installs = app[5].replace('+','')
            n_installs = n_installs.replace(',','')
            install_count += float(n_installs)
    len_genre = frequency_android_category[genre]
    avg_number = install_count/len_genre
    print( 'Average user number in', genre, 'genre is :' ,avg_number )
    avg_install_genre[genre] = avg_number

Average user number in ART_AND_DESIGN genre is : 1986335.0877192982
Average user number in AUTO_AND_VEHICLES genre is : 647317.8170731707
Average user number in BEAUTY genre is : 513151.88679245283
Average user number in BOOKS_AND_REFERENCE genre is : 8767811.894736841
Average user number in BUSINESS genre is : 1712290.1474201474
Average user number in COMICS genre is : 817657.2727272727
Average user number in COMMUNICATION genre is : 38456119.167247385
Average user number in DATING genre is : 854028.8303030303
Average user number in EDUCATION genre is : 1833495.145631068
Average user number in ENTERTAINMENT genre is : 11640705.88235294
Average user number in EVENTS genre is : 253542.22222222222
Average user number in FINANCE genre is : 1387692.475609756
Average user number in FOOD_AND_DRINK genre is : 1924897.7363636363
Average user number in HEALTH_AND_FITNESS genre is : 4188821.9853479853
Average user number in HOUSE_AND_HOME genre is : 1331540.5616438356
Average user number in LIBR

In [87]:
avg_install_genre

{'ART_AND_DESIGN': 1986335.0877192982,
 'AUTO_AND_VEHICLES': 647317.8170731707,
 'BEAUTY': 513151.88679245283,
 'BOOKS_AND_REFERENCE': 8767811.894736841,
 'BUSINESS': 1712290.1474201474,
 'COMICS': 817657.2727272727,
 'COMMUNICATION': 38456119.167247385,
 'DATING': 854028.8303030303,
 'EDUCATION': 1833495.145631068,
 'ENTERTAINMENT': 11640705.88235294,
 'EVENTS': 253542.22222222222,
 'FINANCE': 1387692.475609756,
 'FOOD_AND_DRINK': 1924897.7363636363,
 'HEALTH_AND_FITNESS': 4188821.9853479853,
 'HOUSE_AND_HOME': 1331540.5616438356,
 'LIBRARIES_AND_DEMO': 638503.734939759,
 'LIFESTYLE': 1437816.2687861272,
 'GAME': 15588015.603248259,
 'FAMILY': 3695641.8198090694,
 'MEDICAL': 120550.61980830671,
 'SOCIAL': 23253652.127118643,
 'SHOPPING': 7036877.311557789,
 'PHOTOGRAPHY': 17840110.40229885,
 'SPORTS': 3638640.1428571427,
 'TRAVEL_AND_LOCAL': 13984077.710144928,
 'TOOLS': 10801391.298666667,
 'PERSONALIZATION': 5201482.6122448975,
 'PRODUCTIVITY': 16787331.344927534,
 'PARENTING': 5426

Sort the number in descending order

In [88]:
sorted(avg_install_genre.items(), key=operator.itemgetter(1), reverse = True)

[('COMMUNICATION', 38456119.167247385),
 ('VIDEO_PLAYERS', 24727872.452830188),
 ('SOCIAL', 23253652.127118643),
 ('PHOTOGRAPHY', 17840110.40229885),
 ('PRODUCTIVITY', 16787331.344927534),
 ('GAME', 15588015.603248259),
 ('TRAVEL_AND_LOCAL', 13984077.710144928),
 ('ENTERTAINMENT', 11640705.88235294),
 ('TOOLS', 10801391.298666667),
 ('NEWS_AND_MAGAZINES', 9549178.467741935),
 ('BOOKS_AND_REFERENCE', 8767811.894736841),
 ('SHOPPING', 7036877.311557789),
 ('PERSONALIZATION', 5201482.6122448975),
 ('WEATHER', 5074486.197183099),
 ('HEALTH_AND_FITNESS', 4188821.9853479853),
 ('MAPS_AND_NAVIGATION', 4056941.7741935486),
 ('FAMILY', 3695641.8198090694),
 ('SPORTS', 3638640.1428571427),
 ('ART_AND_DESIGN', 1986335.0877192982),
 ('FOOD_AND_DRINK', 1924897.7363636363),
 ('EDUCATION', 1833495.145631068),
 ('BUSINESS', 1712290.1474201474),
 ('LIFESTYLE', 1437816.2687861272),
 ('FINANCE', 1387692.475609756),
 ('HOUSE_AND_HOME', 1331540.5616438356),
 ('DATING', 854028.8303030303),
 ('COMICS', 81765

The top three app categories that has been installed the most are 'COMMUNICATION', 'VIDEO_PLAYERS', 'SOCIAL'. 'GAME' and 'ENTERTAINMENT' are a little bit lower.
Notice that application in social net work genre are in the top three of both market.

Let's see the percentage of user review in each existing application in Social Networking genre.

In [105]:
avg_install_SN ={}
total_install_SN = 0
for app in android_free:
    if app[1] == 'SOCIAL':
        n_installs = app[5].replace('+','')
        n_installs = n_installs.replace(',','')
        total_install_SN += float(n_installs)
        
for app in android_free:
    if app[1] == 'SOCIAL':
        n_installs = app[5].replace('+','')
        n_installs = n_installs.replace(',','')
        percent_install_count = (float(n_installs)*100)/total_install_SN
        #print (app[0], ':' ,percent_install_count ,'%' )
        avg_install_SN[app[0]] = percent_install_count

In [104]:
sorted(avg_install_SN.items(),key=operator.itemgetter(1), reverse = True)

[('Facebook', 18.222032876511694),
 ('Google+', 18.222032876511694),
 ('Instagram', 18.222032876511694),
 ('Facebook Lite', 9.111016438255847),
 ('Snapchat', 9.111016438255847),
 ('Tumblr', 1.8222032876511696),
 ('Pinterest', 1.8222032876511696),
 ('Badoo - Free Chat & Dating App', 1.8222032876511696),
 ('Tango - Live Video Broadcast', 1.8222032876511696),
 ('LinkedIn', 1.8222032876511696),
 ('Tik Tok - including musical.ly', 1.8222032876511696),
 ('BIGO LIVE - Live Stream', 1.8222032876511696),
 ('VK', 1.8222032876511696),
 ('ooVoo Video Calls, Messaging & Stories', 0.9111016438255848),
 ('MeetMe: Chat & Meet New People', 0.9111016438255848),
 ('Zello PTT Walkie Talkie', 0.9111016438255848),
 ('POF Free Dating App', 0.9111016438255848),
 ('SKOUT - Meet, Chat, Go Live', 0.9111016438255848),
 ('TextNow - free text + calls', 0.18222032876511696),
 ('LiveMe - Video chat, new friends, and make money', 0.18222032876511696),
 ('HTC Social Plugin - Facebook', 0.18222032876511696),
 ('Quora', 

The percentage of installed rate in each app are more scatter than IOS. In the mean time there are so many competitor as well.

Let's see the Communication genre

In [106]:
avg_install_CM ={}
total_install_CM = 0
for app in android_free:
    if app[1] == 'COMMUNICATION':
        n_installs = app[5].replace('+','')
        n_installs = n_installs.replace(',','')
        total_install_CM += float(n_installs)
        
for app in android_free:
    if app[1] == 'COMMUNICATION':
        n_installs = app[5].replace('+','')
        n_installs = n_installs.replace(',','')
        percent_install_count = (float(n_installs)*100)/total_install_CM
        #print (app[0], ':' ,percent_install_count ,'%' )
        avg_install_CM[app[0]] = percent_install_count

In [107]:
sorted(avg_install_CM.items(),key=operator.itemgetter(1), reverse = True)

[('WhatsApp Messenger', 9.060510090313125),
 ('Messenger – Text and Video Chat for Free', 9.060510090313125),
 ('Skype - free IM & video calls', 9.060510090313125),
 ('Google Chrome: Fast & Secure', 9.060510090313125),
 ('Gmail', 9.060510090313125),
 ('Hangouts', 9.060510090313125),
 ('Google Duo - High Quality Video Calls', 4.530255045156562),
 ('imo free video calls and chat', 4.530255045156562),
 ('LINE: Free Calls & Messages', 4.530255045156562),
 ('UC Browser - Fast Download Private & Secure', 4.530255045156562),
 ('Viber Messenger', 4.530255045156562),
 ('imo beta free calls and text', 0.9060510090313125),
 ('Android Messages', 0.9060510090313125),
 ('Who', 0.9060510090313125),
 ('GO SMS Pro - Messenger, Free Themes, Emoji', 0.9060510090313125),
 ('Firefox Browser fast & private', 0.9060510090313125),
 ('Messenger Lite: Free Calls & Messages', 0.9060510090313125),
 ('Kik', 0.9060510090313125),
 ('KakaoTalk: Free Calls & Text', 0.9060510090313125),
 ('Opera Mini - fast web browser

The percentage of installed rate in each app are very scatter. However user may spend more time in the app when they talk to Family or friends. Anyway it is a good option still. It depend on how we design to displayed ads on their chat screen.

Let's see the 'VIDEO_PLAYERS' genre.

In [110]:
avg_install_VP ={}
total_install_VP = 0
for app in android_free:
    if app[1] == 'VIDEO_PLAYERS':
        n_installs = app[5].replace('+','')
        n_installs = n_installs.replace(',','')
        total_install_VP += float(n_installs)
        
for app in android_free:
    if app[1] == 'VIDEO_PLAYERS':
        n_installs = app[5].replace('+','')
        n_installs = n_installs.replace(',','')
        percent_install_count = (float(n_installs)*100)/total_install_VP
        #print (app[0], ':' ,percent_install_count ,'%' )
        avg_install_VP[app[0]] = percent_install_count

In [111]:
sorted(avg_install_VP.items(),key=operator.itemgetter(1), reverse = True)

[('YouTube', 25.43408531444765),
 ('Google Play Movies & TV', 25.43408531444765),
 ('MX Player', 12.717042657223825),
 ('Motorola Gallery', 2.543408531444765),
 ('VLC for Android', 2.543408531444765),
 ('Dubsmash', 2.543408531444765),
 ('VivaVideo - Video Editor & Photo Movie', 2.543408531444765),
 ('VideoShow-Video Editor, Video Maker, Beauty Camera', 2.543408531444765),
 ('Motorola FM Radio', 2.543408531444765),
 ('Vote for', 1.2717042657223825),
 ('Vigo Video', 1.2717042657223825),
 ('MiniMovie - Free Video and Slideshow Editor', 1.2717042657223825),
 ('Samsung Video Library', 1.2717042657223825),
 ('LIKE – Magic Video Maker & Community', 1.2717042657223825),
 ('DU Recorder – Screen Recorder, Video Editor, Live', 1.2717042657223825),
 ('KineMaster – Pro Video Editor', 1.2717042657223825),
 ('VMate', 1.2717042657223825),
 ('HD Video Downloader : 2018 Best video mate', 1.2717042657223825),
 ('Ringdroid', 1.2717042657223825),
 ('Video Downloader', 0.2543408531444765),
 ('Video Player A

The top three application take so many percentage which is around 62%. 

As far as we explore from the top three genre, there still be no outstanding one. Let's see the genre that is similar to 'Reference' genre in IOS which is 'BOOKS_AND_REFERENCE' in number 11 in the list. 


In [112]:
avg_install_BR ={}
total_install_BR = 0
for app in android_free:
    if app[1] == 'BOOKS_AND_REFERENCE':
        n_installs = app[5].replace('+','')
        n_installs = n_installs.replace(',','')
        total_install_BR += float(n_installs)
        
for app in android_free:
    if app[1] == 'BOOKS_AND_REFERENCE':
        n_installs = app[5].replace('+','')
        n_installs = n_installs.replace(',','')
        percent_install_count = (float(n_installs)*100)/total_install_BR
        avg_install_BR[app[0]] = percent_install_count

In [113]:
sorted(avg_install_BR.items(),key=operator.itemgetter(1), reverse = True)

[('Google Play Books', 60.02817986887036),
 ('Bible', 6.002817986887036),
 ('Amazon Kindle', 6.002817986887036),
 ('Wattpad 📖 Free Books', 6.002817986887036),
 ('Audiobooks from Audible', 6.002817986887036),
 ('Wikipedia', 0.6002817986887036),
 ('Cool Reader', 0.6002817986887036),
 ('FBReader: Favorite Book Reader', 0.6002817986887036),
 ('HTC Help', 0.6002817986887036),
 ('Moon+ Reader', 0.6002817986887036),
 ('Aldiko Book Reader', 0.6002817986887036),
 ('Al-Quran (Free)', 0.6002817986887036),
 ('Al Quran Indonesia', 0.6002817986887036),
 ("Al'Quran Bahasa Indonesia", 0.6002817986887036),
 ('Quran for Android', 0.6002817986887036),
 ('Dictionary.com: Find Definitions for English Words', 0.6002817986887036),
 ('English Dictionary - Offline', 0.6002817986887036),
 ('NOOK: Read eBooks & Magazines', 0.6002817986887036),
 ('Dictionary', 0.6002817986887036),
 ('Spanish English Translator', 0.6002817986887036),
 ('Dictionary - Merriam-Webster', 0.6002817986887036),
 ('JW Library', 0.60028179

'Google Play Books' has very high rank in this genre which is about 60%. It lead the second place which is 'Bible' so much. There are a lot of competitor as well.

## Conclusions
In conclusion of both market , Social media are popular in both market but the percentage are mostly toward a few famous applications and there are so many competitor as well. Reference genre is less competitor in IOS store and not in too low rank in Google play store. There may be a lot of room for creativity and may have a chance to be outstanding.